src/gpu/SDL_gpu.c

```diff
@@ -827,6 +827,12 @@ SDL_GPUGraphicsPipeline *SDL_CreateGPUGraphicsPipeline(
                 SDL_assert_release(!"Format is not supported for color targets on this device!");
                 return NULL;
             }
+            if (graphicsPipelineCreateInfo->multisample_state.enable_alpha_to_coverage &&
+                (IsIntegerFormat(graphicsPipelineCreateInfo->target_info.color_target_descriptions[i].format)
+                    || IsCompressedFormat(graphicsPipelineCreateInfo->target_info.color_target_descriptions[i].format))) {
+                SDL_assert_release(!"Format is not compatible with alpha-to-coverage!");
+                return NULL;
+            }
```

Comment:
See https://github.com/gpuweb/gpuweb/pull/4585 and https://github.com/WebKit/WebKit/pull/25455:

1. This check should only run for color target 0 (and verify that it is present).
2. It should verify that the target format (once again, only of color target 0) has an alpha channel.
3. Going by https://github.com/WebKit/WebKit/blob/5bf1259c1e97934f2d648f0805fb7538f5b51447/Source/WebGPU/WebGPU/Texture.mm#L1412, the `IsIntegerFormat` part of the check doesn't seem to be quite right: there are many integer formats that do not support blending and floats may support it (f16 always do). This should be checked for actual blending operations too if SDL is going to enforce it at all - alpha-to-coverage needing blending is just a metal-specific edge case.